ts/features/idpay/onboarding/screens/FailureScreen.tsx

```diff
@@ -1,121 +1,171 @@
+import { OnboardingFailureEnum } from "../types/OnboardingFailure";
 import { useOnboardingMachineService } from "../xstate/provider";
 import { selectOnboardingFailure } from "../xstate/selectors";
-import { Body } from "../../../../components/core/typography/Body";
-
-const failurePictures: Record<OnboardingFailureEnum, IOPictograms> = {
-  [OnboardingFailureEnum.GENERIC]: "umbrella",
-  [OnboardingFailureEnum.UNEXPECTED]: "umbrella",
-  [OnboardingFailureEnum.NOT_STARTED]: "hourglass",
-  [OnboardingFailureEnum.ENDED]: "timeout",
-  [OnboardingFailureEnum.NO_BUDGET]: "timeout",
-  [OnboardingFailureEnum.SUSPENDED]: "timeout",
-  [OnboardingFailureEnum.SESSION_EXPIRED]: "timeout",
-  [OnboardingFailureEnum.NO_REQUIREMENTS]: "error",
-  [OnboardingFailureEnum.ON_EVALUATION]: "hourglass",
-  [OnboardingFailureEnum.NOT_ELIGIBLE]: "error",
-  [OnboardingFailureEnum.ONBOARDED]: "fireworks",
-  [OnboardingFailureEnum.UNSUBSCRIBED]: "error"
-};
+import I18n from "../../../../i18n";
 
 const FailureScreen = () => {
   const machine = useOnboardingMachineService();
   const failureOption = useSelector(machine, selectOnboardingFailure);
 
-  const failure = pipe(
-    failureOption,
-    O.getOrElse(() => OnboardingFailureEnum.GENERIC)
+  const defaultCloseAction = React.useMemo(
+    () => ({
+      label: I18n.t("global.buttons.close"),
+      accessibilityLabel: I18n.t("global.buttons.close"),
+      onPress: () => machine.send({ type: "QUIT_ONBOARDING" })
+    }),
+    [machine]
   );
 
-  const isAlreadyOnboarded = failure === OnboardingFailureEnum.ONBOARDED;
-
-  const handleClosePress = () => {
-    machine.send({ type: "QUIT_ONBOARDING" });
-  };
+  const goToInitiativeAction = React.useMemo(
+    () => ({
+      label: I18n.t("idpay.onboarding.failure.button.goToInitiative"),
+      accessibilityLabel: I18n.t(
+        "idpay.onboarding.failure.button.goToInitiative"
+      ),
+      onPress: () => machine.send({ type: "SHOW_INITIATIVE_DETAILS" })
+    }),
+    [machine]
+  );
 
-  const handleNavigateToInitiativePress = () => {
-    machine.send({ type: "SHOW_INITIATIVE_DETAILS" });
-  };
+  const genericErrorProps = React.useMemo<OperationResultScreenContentProps>(
+    () => ({
+      pictogram: "umbrellaNew",
+      title: I18n.t("idpay.onboarding.failure.message.GENERIC.title"),
+      subtitle: I18n.t("idpay.onboarding.failure.message.GENERIC.subtitle"),
+      action: defaultCloseAction
+    }),
+    [defaultCloseAction]
+  );
 
-  const renderCloseButton = () => {
-    if (isAlreadyOnboarded) {
-      return (
-        <ButtonSolid
-          label={I18n.t("idpay.onboarding.failure.button.goToInitiative")}
-          accessibilityLabel={I18n.t(
-            "idpay.onboarding.failure.button.goToInitiative"
-          )}
-          onPress={handleNavigateToInitiativePress}
-          fullWidth={true}
-        />
-      );
+  // TODO add missing pictograms IOBP-176
+  const mapFailureToContentProps = (
```

Comment:
at this point, why not just handle this like this?
```ts
title: I18n.t(
            `idpay.onboarding.failure.message.${failure}.title`
          ),
subtitle: I18n.t(
            `idpay.onboarding.failure.message.${failure}.subtitle`
          ),
action: defaultCloseAction,
pictogram:myErrorPictograms[failure] // or just add it to I18n
```

Reply:
Not all configurations have title and/or subtitle and not alla configurations have the default close action. I tried a solution like the one you suggested but I found it a bit cumbersome.
Also, using a simple switch like this is very easy to mantain and edit if necessary, imho.
Finally, managing pictograms via i18n is not a good idea.